common/v2/services/Store/BalanceService.tsx

```diff
@@ -136,3 +136,22 @@ export const getAllTokensBalancesOfAccount = async (account: StoreAccount, asset
     throw new Error(err);
   }
 };
+
+export const getAccountsTokenBalance = async (accounts: StoreAccount[], tokenContract: string) => {
+  const scanner = getScanner(accounts[0].network.nodes[0]);
+  try {
+    return scanner.getTokenBalances(accounts.map(account => account.address), tokenContract);
+  } catch (err) {
+    throw new Error(err);
+  }
+};
+
+export const getAccountsUnlockVIPStatus = async (accounts: StoreAccount[]) => {
+  return getAccountsTokenBalance(accounts, MYCRYPTO_UNLOCK_CONTRACT_ADDRESS)
+    .then(unlockStatusBalanceMap =>
+      Object.keys(unlockStatusBalanceMap).filter(accountAddress =>
+        unlockStatusBalanceMap[accountAddress].isGreaterThan(new BN(0))
```

Comment:
1. The method name is `...status`. Best to either
    - rename the function to return all the accounts with a valid balance
    - return the `status`: 'pending' | 'valid' etc
    - return a `boolean` if there is an account that is valid.
2. Since we only want to compare the value how about:
   ```
      Object.values(unlockStatusBalanceMap).filter(balance => balance.isGreaterThan(new BN(0)))
   ```
3. Let's add a comment explaining why anything > 0 is valid.

Reply:
2. In the future we may want the addresses for some reason or another, so i extracted this into a function that returns success `boolean`